### PR TITLE
Refactored adapters to use ChainMap

### LINKED ISSUE #110: ChainMap instead of copying and updating dictionaries
As a developer, I have a preference for using Idiomatic Python and want to refactor adapters to use `ChainMap`.

I was rewatching Raymond Hettinger's [Transforming Code into Beautiful, Idiomatic Python](https://youtu.be/OSGv2VnC0go) and realized we should be using [ChainMap](https://docs.python.org/3/library/collections.html#collections.ChainMap) instead of copying and updating dictionaries. This is most prevalent in the adapters that have been written.

Go thru and find all instances of `dict.copy()` and `dict(item)`; refactor to use ChainMap.

### PR DESCRIPTION
**Pull request by dansdevelopments**
Closes #110 

Refactored adapters to use ChainMaps instead of copying and updating dictionaries

### CODE REVIEW

File: busy_beaver/adapters/github.py [hunk 1]
```diff
@@ -50,8 +52,7 @@ def _get_all_items(self, url, *, max_pages=5) -> List[Dict]:
 
         all_items = []
         for page_num in range(1, min(last_page, max_pages) + 1):
-            combined_params = self.params.copy()
-            combined_params.update({"page": page_num})
+            combined_params = ChainMap({"page": page_num}, self.params)
```
Comment (alysivji) [inline busy_beaver/adapters/github.py]: Might make sense to make `self.param` into a ChainMap and then update it as needed.

**Comment by alysivji** [comment]
Hi @dansdevelopments! Thanks so much for your help; we really appreciate your contributions to Busy Beaver!

Do you have time to update this PR?

**Comment by BinSquare** [comment]
@alysivji I'd like to takeover this PR since no-one seems to be assigned anymore?

**Comment by alysivji** [comment]
> @alysivji I'd like to takeover this PR since no-one seems to be assigned anymore?

That would be great. Please let me know if you need a hand!

**Comment by alysivji** [comment]
Closed via #336 

Python 3.9 has a new dictionary merge operator. Thought it would be easier to use that versus ChainMap. Appreciate your creating this PR! Helped me find where to make changes.

**Pull request closed without merging**